docs/PREACT_BADGES.md

```diff
@@ -0,0 +1,328 @@
+
+`validations.min` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical minimum or `string` / `url` length
+
+`validations.max` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical maximum or `string` / `url` length
+
+`validations.regex` - ensures `defaultValue` or the user defined `value` meets a regex definition. Must also provide `validations.regexExplain`
+
+`validations.regexExplain` - required if using `validations.regex`. Describes the regex definition and is displayed as an error message if the regex validation fails
+
+
+```json
+{
+	"parameters": [
+		{
+			"name": "font_size",
+			"type": "array",
+			"label": "Font Size",
+			"description": "Select the badge font size",
+			"defaultValue": "14px",
+			"options": ["14px", "16px", "18px", "20px", "22px", "24px", "26px", "28px", "30px"]
+		},
+		{
+			"name": "prefix",
+			"type": "string",
+			"label": "Prefix Symbol",
+			"description": "Display a prefix before the badge value. Ie. currency symbol",
+			"validations": {
+				"regex": "^[$€]$",
+				"regexExplain": "Only $ or € currency symbols are allowed"
+			}
+		},
+		{
+			"name": "background",
+			"type": "color",
+			"label": "Badge background color",
+			"description": "Select the badge background color",
+			"defaultValue": "#0000ff"
+		},
+		{
+			"name": "link",
+			"type": "url",
+			"label": "Redirect to URL on click",
+			"description": "Redirect to a URL when the badge is clicked"
+		},
+		{
+			"name": "zindex",
+			"type": "integer",
+			"label": "Z-index",
+			"description": "Set a z-index value for the badge",
+			"validations": {
+				"min": -1,
+				"max": 2147483647
+			}
+		},
+		{
+			"name": "opacity",
+			"type": "decimal",
+			"label": "Opacity",
+			"description": "Badge opacity value between 0 and 1.0",
+			"defaultValue": "1.0",
+			"validations": {
+				"min": 0,
+				"max": 1
+			}
+		},
+		{
+			"name": "bold_value",
+			"type": "boolean",
+			"label": "Make value bold",
+			"description": "Should the badge value be bold text?",
+			"defaultValue": "false"
+		},
+		{
+			"name": "show_border",
+			"type": "checkbox",
+			"label": "Show border",
+			"description": "Display a border around the badge",
+			"defaultValue": "false"
+		},
+		{
+			"name": "show_shadow",
+			"type": "toggle",
+			"label": "Show shadow",
+			"description": "Display a shadow behind the badge",
+			"defaultValue": "true"
+		}
+	]
+}
+```
+
+## Custom Badge Locations
+
+Custom Badge Locations can be created and sync to the Searchspring Management Console using the Snapfu CLI. See [Getting Started > Setup](https://searchspring.github.io/snap/#/start-setup) for installing Snapfu.
+
+We can create custom overlay and callout locations by defining a `locations.json` file in the project path: `src/components/Badges/locations.json`
```

Comment:
The locations.json can really exist anywhere though right? Not sure it is worth mentioning though.